src/mapblock.h

```diff
@@ -292,46 +277,23 @@ class MapBlock
 	//// Non-checking variants of the above
 	////
 
-	inline MapNode getNodeNoCheck(s16 x, s16 y, s16 z, bool *valid_position)
-	{
-		*valid_position = data != nullptr;
-		if (!*valid_position)
-			return {CONTENT_IGNORE};
-
-		return data[z * zstride + y * ystride + x];
-	}
-
-	inline MapNode getNodeNoCheck(v3s16 p, bool *valid_position)
-	{
-		return getNodeNoCheck(p.X, p.Y, p.Z, valid_position);
-	}
-
-	////
-	//// Non-checking, unsafe variants of the above
-	//// MapBlock must be loaded by another function in the same scope/function
-	//// Caller must ensure that this is not a dummy block (by calling isDummy())
-	////
-
-	inline const MapNode &getNodeUnsafe(s16 x, s16 y, s16 z)
+	inline MapNode getNodeNoCheck(s16 x, s16 y, s16 z)
 	{
 		return data[z * zstride + y * ystride + x];
 	}
 
-	inline const MapNode &getNodeUnsafe(v3s16 &p)
+	inline MapNode getNodeNoCheck(v3s16 p)
```

Comment:
I am a bit dubious about these.
Yes, a v3s16 technically only uses 48bit. Is it really faster constructing a new object compared to passing a (64bit) reference?

I feel like we went through a while back and just changes things the other way.

Reply:
In this case it probably doesn't matter either way, since the function is inline. But structs of this size can usually be passed directly in registers, I believe.

Reply:
FWIW, I compiled the following snippet with g++ (default flags, `-O0`) and found that v3s16 is finally completely shifted into `RSI` before calling `getNodeNoCheck`. Passing `v3s16` by reference results in a `LEA` command but less shifting & binary OR operations. They're both indeed very close.

https://gist.github.com/SmallJoker/4027d7f33a424d1cdc688d651af77e37

Reply:
Here's a comparison with -O3: https://godbolt.org/z/P3G8dcKEE